File: Problem_03/AC_solution.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        string s;
        cin >> s;
        int ans;
        int lastOne = -1, firstZero = s.length();
        for (int i = 0; i < s.length(); i++)
        {
            if (s[i] == '1')
                lastOne = i;
            if (s[i] == '0')
            {
                firstZero = i;
                break;
            }
        }
        if (lastOne != -1 && firstZero != s.length())
            ans = firstZero - lastOne + 1;
        else if (lastOne == -1 && firstZero == s.length())
        {
            ans = firstZero;
        }
        else
        {
            ans = firstZero - lastOne;
        }
        cout << ans << endl;
    }
    return 0;
}

File: Problem_03/AC_solution.py
t = int(input())

for _ in range(t):
    s = input()
    ans = 0
    last_one = -1
    first_zero = len(s)

    for i in range(len(s)):
        if s[i] == '1':
            last_one = i
        if s[i] == '0':
            first_zero = i
            break

    if last_one != -1 and first_zero != len(s):
        ans = first_zero - last_one + 1
    elif last_one == -1 and first_zero == len(s):
        ans = first_zero
    else:
        ans = first_zero - last_one

    print(ans)
